### PR TITLE
Fix some compiler warnings

### PR DESCRIPTION
**Pull request by vedgy**
See the commit messages for details.

### CODE REVIEW

File: YACReaderLibrary/server/controllers/v2/pagecontroller_v2.cpp [hunk 1]
```diff
@@ -33,7 +31,6 @@ void PageControllerV2::service(HttpRequest &request, HttpResponse &response)
     bool remote = path.endsWith("remote");
 
     QStringList pathElements = path.split('/');
-    QString libraryName = DBHelper::getLibraryName(pathElements.at(2).toInt());
```
Comment (vedgy) [inline YACReaderLibrary/server/controllers/v2/pagecontroller_v2.cpp:36]: This and the identical change in `PageController::service()` are the ones I'm not sure about. If the call to `DBHelper::getLibraryName()` is useful here, either the variable assignment should be removed or an explaining comment should be added.

**Comment by selmf** [comment]
From a first glance, this is a bit of a mixed bag. A good amount of these are no-brainers and can be easily merged but there are some commit thrown in (like the pagecontroller) that need closer examination which I can't provide given that I don't use YACReader for iOS.
@vedgy can I trouble you to put the commits that need examination into a separate PR? I'm currently working on something that requires the removal of deprecated api usage and it would be stupid to duplicate your work.

**Comment by vedgy** [comment]
I think the only risky commit here is the `PageController` one. I can move that in a separate pull request.

> I'm currently working on something that requires the removal of deprecated api usage and it would be stupid to duplicate your work.

I've also ported away from some deprecated Qt API in #206, so you may want to look at it as well.

**Comment by vedgy** [comment]
Split the `PageController` commit into #231.

**Pull request merged**